compiler/rustc_target/src/target_features.rs

```diff
@@ -192,6 +192,11 @@ const X86_ALLOWED_FEATURES: &[(&str, Stability)] = &[
     // tidy-alphabetical-start
     ("adx", Stable),
     ("aes", Stable),
+    ("amx-bf16", Stable),
+    ("amx-complex", Stable),
+    ("amx-fp16", Stable),
+    ("amx-int8", Stable),
+    ("amx-tile", Stable),
```

Comment:
these should not be Stable. please create a new tracking issue for a `feature(amx_target_feature)`, and follow https://rustc-dev-guide.rust-lang.org/implementing_new_features.html#stability-in-code for making these Unstable but available based on that feature.

Reply:
I get the unstability for AMX, why should the SHA512 ones be unstable? These are extremely simple intrinsics with ~6 intrinsics each.

Reply:
If you want those reviewed separately then you would want to open them as a separate PR and make your case for an FCP for insta-stabilizing them.

However, the fact that the intrinsics aren't even public in `core::arch` with this PR, but that such depends on another PR, due to our somewhat questionable organizational decisions, suggests we should at least wait on stabilizing them until we can test that the publicly-available intrinsics codegen correctly.

Reply:
right, they should go together.

Reply:
sure! it will make it easier to review the implementation history at least.